### ISSUE TITLE
[BUG?] Failure to decompile on Linux

### ISSUE THREAD

**Issue by FezzedOne** [UnrealMultiple/TerraAngel#4]
Here's the output from `./fast.ps1 -decompile`:

```
Decompiling steam/Terraria/Terraria.exe into src/Terraria
Version not supported: 4.0.0.0
System.NotSupportedException: Version not supported: 4.0.0.0
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.GetMscorlibBasePath(Version version, String publicKeyToken) in /_/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs:line 609
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.GetCorlib(IAssemblyReference reference) in /_/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs:line 543
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.ResolveInternal(IAssemblyReference name) in /_/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs:line 448
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.FindAssemblyFile(IAssemblyReference name) in /_/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs:line 297
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.Resolve(IAssemblyReference name) in /_/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs:line 214
   at DotnetPatcher.Decompile.EmbeddedAssemblyResolver.Resolve(IAssemblyReference name) in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/DotnetPatcher/DotnetPatcher/Decompile/EmbeddedAssemblyResolver.cs:line 48
   at DotnetPatcher.Decompile.EmbeddedAssemblyResolver.<>c__DisplayClass6_0.<ResolveAsync>b__0() in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/DotnetPatcher/DotnetPatcher/Decompile/EmbeddedAssemblyResolver.cs:line 62
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at ICSharpCode.Decompiler.TypeSystem.DecompilerTypeSystem.InitializeAsync(PEFile mainModule, IAssemblyResolver assemblyResolver, TypeSystemOptions typeSystemOptions) in /_/ICSharpCode.Decompiler/TypeSystem/DecompilerTypeSystem.cs:line 275
   at ICSharpCode.Decompiler.TypeSystem.DecompilerTypeSystem..ctor(PEFile mainModule, IAssemblyResolver assemblyResolver, TypeSystemOptions typeSystemOptions) in /_/ICSharpCode.Decompiler/TypeSystem/DecompilerTypeSystem.cs:line 227
   at ICSharpCode.Decompiler.TypeSystem.DecompilerTypeSystem..ctor(PEFile mainModule, IAssemblyResolver assemblyResolver, DecompilerSettings settings) in /_/ICSharpCode.Decompiler/TypeSystem/DecompilerTypeSystem.cs:line 217
   at DotnetPatcher.Utility.DecompilerUtility.DecompileModule(PEFile module, IAssemblyResolver resolver, ExtendedProjectDecompiler decompiler, List`1 items, ISet`1 sourceSet, ISet`1 resourceSet, String projectOutputDirectory, DecompilerSettings settings, ICollection`1 exclude, String conditional) in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/DotnetPatcher/DotnetPatcher/Utility/DecompilerUtility.cs:line 217
   at DotnetPatcher.Utility.ProjectFileUtility.AddEmbeddedLibrary(Resource res, String projectOutputDirectory, ExtendedProjectDecompiler decompiler, DecompilerSettings settings, IAssemblyResolver resolver, List`1 items) in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/DotnetPatcher/DotnetPatcher/Utility/ProjectFileUtility.cs:line 145
   at DotnetPatcher.Decompile.Decompiler.Decompile(String[] decompiledLibraries) in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/DotnetPatcher/DotnetPatcher/Decompile/Decompiler.cs:line 81
   at Program.DecompileTerraria() in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/TerraAngelSetup/Program.cs:line 225
   at Program.Main(String[] args) in /home/renofezzedone/.git/TerraAngel/TerraAngelSetup/TerraAngelSetup/Program.cs:line 159
Failed. Open an issue and attach the output.
```

I'm on up-to-date Arch. Here are my .NET runtime and PowerShell package versions:

```
 ~ > pacman -Q | grep 'dotnet'               
dotnet-host 8.0.8.sdk108-1
dotnet-runtime 8.0.8.sdk108-1
dotnet-runtime-6.0 6.0.33.sdk133-1
dotnet-sdk 8.0.8.sdk108-1
dotnet-targeting-pack 8.0.8.sdk108-1
 ~ > pacman -Q | grep 'powershell'
powershell-bin 7.4.6-1
```

**Comment by LaoSparrow** [comment]
![image](https://github.com/user-attachments/assets/43686542-48a9-4aa4-99b2-d87427488050)
you have to put dotnet framework files "Microsoft.NET" under the project root directory (sorry I forgot to mention that)
It is located under C:\Windows if you have installed dotnet framework
or you can get it from a wine prefix that have dotnet framework installed (just like what i did)

**Comment by LaoSparrow** [comment]
resolved by commit 9eeb16d86bc80e1bed35579117d3718ad366343e

**Comment by FezzedOne** [comment]
I know I'm necroing by now, but thanks.
